android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -255,105 +267,93 @@ public boolean hasRequiredPermissions() {
     }
 
     /**
-     * Exported plugin call for checking the granted status for each permission
-     * declared on the plugin. This plugin call responds with a mapping of permissions to
-     * the associated granted status.
+     * Helper for requesting a specific permission
      *
      * @since 3.0.0
+     * @param call the plugin call
+     * @param permission the permission to request
+     * @param requestCode the requestCode to use to associate the result with the plugin
      */
-    @PluginMethod
-    public void checkPermissions(PluginCall pluginCall) {
-        JSObject permissionsResult = getPermissionStates();
-        pluginCall.resolve(permissionsResult);
+    public void requestPermission(PluginCall call, String permission, int requestCode) {
```

Comment:
Is this intended for plugins devs to use?

Reply:
Yes, its a refactor of the existing `pluginRequestPermission` that includes the savePermissionCall